src/map/script.c

```diff
@@ -15785,19 +15785,32 @@ BUILDIN(getmapxy)
 	return true;
 }
 
+enum logmes_type {
+	LOGMES_NPC,
+	LOGMES_ATCOMMAND
+};
+
 /*==========================================
- * Allows player to write NPC logs (i.e. Bank NPC, etc) [Lupus]
+ * Allows player to write logs (i.e. Bank NPC, etc) [Lupus]
  *------------------------------------------*/
-BUILDIN(logmes)
-{
-	const char *str;
+BUILDIN(logmes) {
+	const char *str = script_getstr(st, 2);
 	struct map_session_data *sd = script->rid2sd(st);
+	enum logmes_type type = LOGMES_NPC;
+	nullpo_retr(sd, false);
 
-	if (sd == NULL)
-		return true;
+	if (script_hasdata(st, 3)) {
+		type = script_getnum(st, 3);
+	}
+
+	switch (type) {
+	case LOGMES_ATCOMMAND:
+		logs->atcommand(sd, str);
+		break;
+	case LOGMES_NPC:
+		logs->npc(sd, str);
+	}
```

Comment:
How about adding a `default:` here that prints an error message? (so that a wrong value won't go silently unnoticed)

Reply:
good idea